package/MDAnalysis/coordinates/FHIAIMS.py

```diff
@@ -0,0 +1,333 @@
+       velocities if given, all according to the FHIAIMS-format
+       specifications https://doi.org/10.6084/m9.figshare.12413477.v1
+       See also https://aimsclub.fhi-berlin.mpg.de/
+
+    """
+    format = ['IN', 'FHIAIMS']
+    units = {'time': 'ps', 'length': 'Angstrom', 'velocity': 'Angstrom/ps'}
+    _Timestep = Timestep
+
+    def _read_first_frame(self):
+        with util.openany(self.filename, 'rt') as fhiaimsfile:
+            relative, positions, velocities, lattice_vectors = [], [], [], []
+            skip_tags = ["#", "initial_moment"]
+            oldline = ''
+            for line in fhiaimsfile:
+                line = line.strip()
+                if line.startswith("lattice"):
+                    lattice_vectors.append(line.split()[1:])
+                    oldline = line
+                    continue
+                if line.startswith("atom"):
+                    positions.append(line.split()[1:-1])
+                    relative.append('atom_frac' in line)
+                    oldline = line
+                    continue
+                if line.startswith("velocity"):
+                    if not 'atom' in oldline:
+                        raise ValueError(
+                            'Non-conforming line (velocity must follow atom): ({0})in FHI-AIMS input file {0}'.format(line, self.filename))
+                    velocities.append(line.split()[1:])
+                    oldline = line
+                    continue
+                if any([line.startswith(tag) for tag in skip_tags]):
+                    oldline = line
+                    continue
+                raise ValueError(
+                    'Non-conforming line: ({0})in FHI-AIMS input file {0}'.format(line, self.filename))
+
+            lattice_vectors = np.asarray(lattice_vectors, dtype=np.float32)
+            positions = np.asarray(positions, dtype=np.float32)
+            velocities = np.asarray(velocities, dtype=np.float32)
+            relative = np.asarray(relative)
+
+            if len(velocities) not in [0, len(positions)]:
+                raise ValueError(
+                    'Found incorrect number of velocity tags in the FHI-AIMS file.')
+            if len(lattice_vectors) not in [0, 3]:
+                raise ValueError(
+                    'Found partial periodicity in FHI-AIMS file. This cannot be handled by MDAnalysis.')
+            if len(lattice_vectors) == 0 and any(relative):
+                raise ValueError(
+                    'Found relative coordinates in FHI-AIMS file without lattice info.')
+
+            self.n_atoms = n_atoms = len(positions)
+            self.ts = ts = self._Timestep(n_atoms, **self._ts_kwargs)
+
+            if len(lattice_vectors) > 0:
+                positions[relative] = np.matmul(
+                    positions[relative], lattice_vectors)
+                self.ts._unitcell[:] = lattice_vectors
+
+            ts._pos = positions.copy()
```

Comment:
I don't think we need to make a copy here because the `_pos` setter fills the array anyway. Avoiding the copy should improve performance. — @richardjgowers ?

Reply:
Er, sorry, `ts._pos` is the actual array. If we used
```
ts.positions = positions
```
as we do in, for instance, in the XYZReader, then we fill the array. The latter is cleaner and makes sure that the dtype stays. We also don't need the asarray casting above, that's done with the assignment to a np array of a given dtype.

Reply:
Tried `ts.positions = positions` and it works. Will push my changes.

Reply:
> I'll look at the docs and make any other changes.

Thanks!